internal/adapter/provider/cliproxyapi_grok/adapter_test.go

```
@@ -159,6 +160,81 @@ func TestEnsureOpenAIStreamFinishBeforeDoneDoesNotDuplicateFinishReason(t *testi
+			RefreshToken: "refresh-token",
+			BaseURL:      upstream.URL,
+		}},
+	}
+	adapter, err := NewAdapter(provider)
+	if err != nil {
+		t.Fatalf("NewAdapter() error = %v", err)
+	}
+	grok := adapter.(*CLIProxyAPIGrokAdapter)
+
+	body := []byte(`{"model":"grok-4","messages":[{"role":"user","content":"say ok"}]}`)
+	req := httptest.NewRequest(http.MethodPost, "/v1/chat/completions", strings.NewReader(string(body)))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)